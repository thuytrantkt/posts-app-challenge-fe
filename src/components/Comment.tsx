import React from "react";
import { CommentType } from "../types";

const Comment = ({
  comment,
  firstComment,
}: {
  comment: CommentType;
  firstComment: boolean;
}) => {
  return (
    <div
      style={{
        borderTop: !firstComment ? "1px solid #ccc" : "none",
        padding: "10px 0",
        fontSize: "0.9em",
      }}
    >
      <p>
        {comment.content} - <i>{comment.author}</i>,{" "}
        {new Date(comment.createdAt).toLocaleDateString()}
      </p>
    </div>
  );
};

export default Comment;
