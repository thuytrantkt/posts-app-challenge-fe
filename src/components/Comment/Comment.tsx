import React from "react";
import { CommentType } from "../../types";
import { dateFormatted } from "../../utils/date-helper";

const Comment = ({
  comment,
  firstComment,
}: {
  comment: CommentType;
  firstComment: boolean;
}) => {
  const { content, author, createdAt } = comment;
  return (
    <div
      style={{
        borderTop: !firstComment ? "1px solid #ccc" : "none",
        padding: "10px 0",
        fontSize: "0.9em",
      }}
    >
      <p>
        {content} - <i>{author}</i>, {dateFormatted(createdAt)}
      </p>
    </div>
  );
};

export default Comment;
