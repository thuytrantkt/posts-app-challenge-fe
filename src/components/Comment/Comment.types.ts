import { CommentType } from "../../types";

export type CommentProps = {
  comment: CommentType;
  firstComment: boolean;
};
