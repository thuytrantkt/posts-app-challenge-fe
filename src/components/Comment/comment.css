.comment-container {
  padding: 10px 0;
  font-size: 0.9em;
}

.border-line {
  border-top: 1px solid #ccc;
}
