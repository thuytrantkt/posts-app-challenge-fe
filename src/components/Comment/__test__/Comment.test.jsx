import { render } from "@testing-library/react";
import Comment from "../Comment";

it("renders Comment component properly without the border line if this is the first comment", () => {
  const commentMockData = {
    author: "John Smith",
    content: "Great post, really made me think.",
    createdAt: "Thu, 15 Feb 2024 17:40:30 GMT",
    id: 1,
    postId: 1,
  };
  const view = render(<Comment comment={commentMockData} firstComment />);
  expect(view).toMatchSnapshot();
});

it("renders Comment component properly with the border line if this is not the first comment", () => {
  const commentMockData = {
    author: "John Smith",
    content: "Great post, really made me think.",
    createdAt: "Thu, 15 Feb 2024 17:40:30 GMT",
    id: 1,
    postId: 1,
  };
  const view = render(
    <Comment comment={commentMockData} firstComment={false} />
  );
  expect(view).toMatchSnapshot();
});
