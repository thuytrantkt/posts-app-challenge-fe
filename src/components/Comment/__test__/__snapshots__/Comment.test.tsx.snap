// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Comment component properly with the border line if this is not the first comment 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="border-line"
      >
        <p>
          Great post, really made me think.
           - 
          <i>
            John Smith
          </i>
          , 
          2/15/2024
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="border-line"
    >
      <p>
        Great post, really made me think.
         - 
        <i>
          John Smith
        </i>
        , 
        2/15/2024
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders Comment component properly without the border line if this is the first comment 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
      >
        <p>
          Great post, really made me think.
           - 
          <i>
            John Smith
          </i>
          , 
          2/15/2024
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
    >
      <p>
        Great post, really made me think.
         - 
        <i>
          John Smith
        </i>
        , 
        2/15/2024
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
